templates/agent/src/components/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import {useTheme} from 'next-themes'
+import {useEffect, useState} from 'react'
+
+export function ThemeSwitcher() {
+	const {theme, setTheme} = useTheme()
+
+	const [mounted, setMounted] = useState(false)
+
+	useEffect(() => {
+		setMounted(true)
+	}, [])
```

Comment:
Interesting - how does the toggle behave without this?

Reply:
this is a habit, toggle has to work on client only, this is the most reliable way to ensure we are on client, that being said I haven't tried using theme switchers without this hack in a while, will test

Reply:
gotta love nextjs

<img width="940" alt="CleanShot 2023-11-03 at 21 44 12@2x" src="https://github.com/RubricLab/create-rubric-app/assets/138781964/91f5b550-1bb2-4325-9cc1-033281ff1434">
<img width="950" alt="CleanShot 2023-11-03 at 21 44 41@2x" src="https://github.com/RubricLab/create-rubric-app/assets/138781964/90c70adf-5708-4afb-a7a2-de5be90a29e5">

Reply:
Classic. After swapping the SVG for Lucide imports (a7e4477), do we still see an error?

Reply:
i don't get any errors regardless if have svg or lucide imports now 😂

I guess I can remove that hack first? unable to get hydration errors now, running dev with bun

Reply:
removed cause unable to reproduce the error so far, worth keeping a note just in case